tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+    rados_obj.run_ceph_command(cmd=config_cmd)
+    time.sleep(5)
+
+    if config.get("online_command_verification"):
+        log.info(
+            "Verifying the usage of online commands to change the acting set of the pool"
+        )
+        for pool_name in existing_pools:
+            pool_details = rados_obj.get_pool_details(pool=pool_name)
+            log.debug(f"Selected pool name : {pool_name}")
+
+            # Testing on replicated pools
+            if not pool_details["erasure_code_profile"]:
+                pool = pool_details["pool_name"]
+
+                # Fetching the acting set for the pool
+                # Getting a sample PG from the pool
+                pool_id = pool_obj.get_pool_id(pool_name=pool)
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+
+                log.debug(f"Acting set for PG-id {pgid} in pool {pool} is {acting_set}")
+
+                # Trying to change the primary with existing primary. -ve test
+                new_osd = acting_set[0]
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = (
+                    f"Error EINVAL: osd.{new_osd} is already primary for pg {pgid}"
+                )
+
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.debug(
+                        "Could not change the primary with the existing primary. Pass"
+                    )
+                    log.error(f"An error occurred but was expected: {e}")
+
+                # Trying to set a non-existant OSD as for the PG primary. -ve test
+                new_osd = 250
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = f"Error ENOENT: osd.{new_osd} does not exist"
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.error(f"An error occurred but was expected: {e}")
+                    log.debug(
+                        "Could not change the primary with non existent OSD. Pass"
+                    )
```

Comment:
As an enhancement later on, maybe we can modularize this section of the code, and provide the error string and command output as inputs

Reply:
Noted.